v2/tests/logger_perf/main.c

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <string.h>
+#include <stdio.h>
+#include <inttypes.h>
+
+#include "macro_utils/macro_utils.h"
+
+#include "timer.h"
+
+#include "c_logging/logger.h"
+#include "c_logging/xlogging.h"
+
+#define TEST_TIME 5000 // ms
+
+#define PERF_TEST_LIST_VALUES \
+    PERF_TEST_LOG_CRITICAL
+
+MU_DEFINE_ENUM_WITHOUT_INVALID(PERF_TEST_LIST, PERF_TEST_LIST_VALUES);
+MU_DEFINE_ENUM_STRINGS_WITHOUT_INVALID(PERF_TEST_LIST, PERF_TEST_LIST_VALUES);
+
+typedef struct PERF_RESULT_TAG
+{
+    double time;
+    uint64_t log_count;
+} PERF_RESULT;
+
+PERF_RESULT v1_results[MU_ENUM_VALUE_COUNT_WITHOUT_INVALID(PERF_TEST_LIST)];
+PERF_RESULT v2_results[MU_ENUM_VALUE_COUNT_WITHOUT_INVALID(PERF_TEST_LIST)];
+PERF_RESULT v2_results_local_context_once[MU_ENUM_VALUE_COUNT_WITHOUT_INVALID(PERF_TEST_LIST)];
+PERF_RESULT v2_results_dynamic_context_once[MU_ENUM_VALUE_COUNT_WITHOUT_INVALID(PERF_TEST_LIST)];
+PERF_RESULT v2_results_local_context_every_time[MU_ENUM_VALUE_COUNT_WITHOUT_INVALID(PERF_TEST_LIST)];
+PERF_RESULT v2_results_dynamic_context_every_time[MU_ENUM_VALUE_COUNT_WITHOUT_INVALID(PERF_TEST_LIST)];
+
+/*not much in the way of testing, but "should not crash" */
+int main(void)
```

Comment:
>main [](http://example.com/codeflow?start=4&length=4)

I think the measurements of these tests should be achieved with at least 1 session and 1 consumer connected to ehe ETW frameowrk. otherwise, ETW is very smart about dropping unwanted packets.

... also tests on the console /console + etw / etw only should have different numbers

Reply:
Agree, will do that on my next measurements